### PR TITLE
feat(compiler): set `enableLegacyTemplate` to false by default

### PR DESCRIPTION
**Pull request by ocombe**
## PR Type
What kind of change does this PR introduce?

```
[x] Feature
```

## What is the current behavior?
The compiler option `enableLegacyTemplate` is enabled by default

## What is the new behavior?
The compiler option `enableLegacyTemplate` is now disabled by default as `<template>` elements have been deprecated since v4. Use `<ng-template>` instead.

## Does this PR introduce a breaking change?
```
[x] Yes
```

### CODE REVIEW

**Review by vicb — APPROVED** [review]
Please add a note in the commit message that this seeting will be removed in v6.

**Comment by ocombe** [comment]
Hmm I don't know why this test is failing :-/
I've tried it locally, the test fails but when I test the example it works fine manually

**Comment by ocombe** [comment]
@juleskremer can you validate the changes in the structural directives guide/examples please? https://pr18756-e255017.ngbuilds.io/guide/structural-directives

File: aio/content/examples/ngcontainer/src/app/app.component.html [hunk 1]
```diff
@@ -105,9 +105,9 @@
     The <code>hero.id</code> in the &lt;template [ngIf]&gt;
```
Comment (vicb) [inline aio/content/examples/ngcontainer/src/app/app.component.html]: `ng-template` (serach and replace `&lt;template`)

File: aio/content/guide/structural-directives.md [hunk 2]
```diff
@@ -208,17 +208,7 @@ Here is `*ngIf` displaying the hero's name if `hero` exists.
 
 
 The asterisk is "syntactic sugar" for something a bit more complicated.
-Internally, Angular desugars it in two stages.
```
Comment (vicb) [inline aio/content/guide/structural-directives.md]: sounds wrong.

I agree we should remove the "desugars it in two stages." it is wrong.

File: aio/content/guide/structural-directives.md [hunk 3]
```diff
@@ -208,17 +208,7 @@ Here is `*ngIf` displaying the hero's name if `hero` exists.
...
-
-
-<code-example path="structural-directives/src/app/app.component.html" linenums="false" title="src/app/app.component.html (ngif-template-attr)" region="ngif-template-attr">
-
-</code-example>
-
-
-
-Then it translates the template _attribute_ into a `<ng-template>` _element_, wrapped around the host element, like this.
+Internally, Angular translates the template _attribute_ into a `<ng-template>` _element_, wrapped around the host element, like this.
```
Comment (vicb) [inline aio/content/guide/structural-directives.md]: `Angular translates the template _attribute_` this is wrong - this attribute is no more significant for Angular

**Comment by vicb** [comment]
Agree with the code changes.

Not sure about the aio changes:
- added some comments in the PR,
- image resolution looks very different, is this  expected ?

**Review by juleskremer — COMMENTED** [review]
LGTM with Vic changes; also, we need to check the sizing on the images.

File: aio/content/guide/structural-directives.md (hunk 3 above)
Comment (ocombe) [inline aio/content/guide/structural-directives.md]: the guide is talking about the *ngIf attribute actually, I'll change the phrasing

**Review by vicb — APPROVED** [review]

**Review by IgorMinar — APPROVED** [review]
Can we change this to a "feat" rather than "refactor" - since we are changing the behavior, we should not claim that this change is a refactor. refactorings should have no visible impact for external developers.

**Comment by ocombe** [comment]
@IgorMinar done

**Comment by jasonaden** [comment]
@ocombe Please have this pass Travis, then we can merge.

**Pull request closed without merging**